script.js

```diff
@@ -0,0 +1,360 @@
+
+function getSquaredNumbers(arrayOfNumbers) {
+    return arrayOfNumbers.map(
+        function (number) {
+            return number ** 2;
+        }
+    )
+}
+
+console.log(getSquaredNumbers([1, 2, 3, 4, 5])); // [1, 4, 9, 16, 25]
+console.log(getSquaredNumbers([6, 7, 8, 9, 10])); // [36, 49, 64, 81, 100]
+
+const numbers = [1, 2, 3];
+const squaredNumbers = getSquaredNumbers(numbers);
+console.log(squaredNumbers); // [1, 4, 9]
+console.log(numbers); // [1, 2, 3]
+
+// 4. Write the getReversedString function.
+
+function getReversedString(string) {
+    return string.split('').reverse().join('');
+}
+
+console.log(getReversedString('Hello!')); // '!olleH'
+console.log(getReversedString('Arrays')); // 'syarrA'
+
+// 5. Write the isPalindrome function.
+// Inside the isPalindrome function, call the getReversedString function.
+// It should be case-insensitive and shouldn't take spaces into consideration.
+
+function isPalindrome(string) {
+    const replaceCharacters = string.replace(/\s/g, '').toLowerCase();
+    const checkStringReverse = getReversedString(replaceCharacters);
+    return checkStringReverse === replaceCharacters;
+}
+
+
+console.log(isPalindrome('Kayak')); // true
+console.log(isPalindrome('Racecar')); // true
+console.log(isPalindrome('Was it a cat I saw')); // true
+console.log(isPalindrome('Hello!')); // false
+
+// 6. Write the countLetters function.
+// It should return an object representing the number of letters in the given text.
+// Ignore spaces and make sure your function is case-insensitive.
+
+function countLetters(text) {
+    const lettersCount = {};
+    const processedText = text.replace(/\s/g, '').toLowerCase();
+    for (let i = 0; i < processedText.length; i++) {
+        const character = processedText[i];
+        if (/[a-z]/.test(character)) {
+            if (lettersCount[character]) {
+                lettersCount[character]++;
+            } else {
+                lettersCount[character] = 1;
+            }
+        }
+    }
+    return lettersCount;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7.Write the getAgeDifference function that finds the difference between the youngest and oldest person.
+// Inside of the getAgeDifference function call the getYoungestPerson  and getOldestPerson functions that you write yourself.
+
+const peopleArray = [
+  {
+    name: 'Adam',
+    age: 20
+  },
+  {
+    name: 'Amanda',
+    age: 5
+  },
+  {
+    name: 'John',
+    age: 75
+  },
+  {
+    name: 'Dave',
+    age: 15
+  }
+]
+
+console.log(getAgeDifference(peopleArray)); // 70
+
+function getYoungestPerson(people) {
+    const userSortedByAge = people.slice().sort(
+        function (firstUser, secondUser) {
+            return firstUser.age - secondUser.age;
+        }
+    )
+    return userSortedByAge[0];
+}
+
+function getOldestPerson(people) {
+    const userSortedByAge = people.slice().sort(
```

Comment:
This is an array with multiple users, so it should be `usersSortedByAge`, not `userSortedByAge`.